blake3/src/lib.rs

```diff
@@ -35,3 +35,35 @@ impl CryptographicHasher<u8, [u8; 32]> for Blake3 {
         hasher.finalize().into()
     }
 }
+
+impl CryptographicHasher<u32, [u32; 8]> for Blake3 {
```

Comment:
I think that we should have another struct `Blake3zk` or something that uses our precompiles, it could even be in the sp1 recursion crate. The reason is that we would eventually want the normal Blake3 to implement hashing on `u32` (this might be useful for SIMD utilization. of the prover)